pyintegration/tests/test_datetimeutils.py

```diff
@@ -0,0 +1,227 @@
+#
+#   Copyright (c) 2016-2021 Deephaven Data Labs and Patent Pending
+#
+
+import unittest
+from time import sleep
+
+from deephaven2.constants import NULL_LONG
+from deephaven2.datetimeutils import *
+
+
+class DateTimeUtilsTestCase(unittest.TestCase):
+    def test_convert_datetime(self):
+        datetime_str = "2021-12-10T23:59:59"
+        timezone_str = "NY"
+        dt = convert_datetime(f"{datetime_str} {timezone_str}")
+        self.assertTrue(str(dt).startswith(datetime_str))
+
+        with self.assertRaises(DHError) as cm:
+            datetime_str = "2021-12-10T23:59:59"
+            timezone_str = "--"
+            dt = convert_datetime(f"{datetime_str} {timezone_str}")
+        self.assertIn("RuntimeException", str(cm.exception))
+
+    def test_convert_period(self):
+        period_str = "1W"
+        period = convert_period(period_str)
+        self.assertEqual(str(period).upper(), period_str)
+
+        period_str = "T1M"
+        period = convert_period(period_str)
+        self.assertEqual(repr(period).upper(), period_str)
+
+        with self.assertRaises(DHError) as cm:
+            period_str = "T1Y"
+            period = convert_period(period_str)
+        self.assertIn("RuntimeException", str(cm.exception))
+
+    def test_convert_time(self):
```

Comment:
looks like this is the function that creates nanos from a string.  Does it do anything else?

Reply:
no